middlewares/hacker.middleware.js

```diff
@@ -112,12 +117,142 @@ async function downloadResume(req, res, next) {
+            };
+            break;
+
+        case Constants.HACKER_STATUS_CANCELLED:
+            //send cancelled email
+            mailData = {
+                to: req.email,
+                from: process.env.NO_REPLY_EMAIL,
+                subject: "Sorry to see you go.",
+                html: fs.readFileSync(path.join(__dirname, "../assets/email/cancelled.html")).toString()
+            };
+            break;
+
+        case Constants.HACKER_STATUS_CHECKED_IN:
+            //send checked in email
+            mailData = {
+                to: req.email,
+                from: process.env.NO_REPLY_EMAIL,
+                subject: `Welcome to ${process.env.HACKATHON}!`,
+                html: fs.readFileSync(path.join(__dirname, "../assets/email/checkedIn.html")).toString()
+            };
+            break;
+
+        case Constants.HACKER_STATUS_CONFIRMED:
+            //send confirmed email
+            mailData = {
+                to: req.email,
+                from: process.env.NO_REPLY_EMAIL,
+                subject: "Thanks for applying!",
+                html: fs.readFileSync(path.join(__dirname, "../assets/email/confirmed.html")).toString()
+            };
+            break;
+
+        case Constants.HACKER_STATUS_WAITLISTED:
+            mailData = {
+                to: req.email,
+                from: process.env.NO_REPLY_EMAIL,
+                subject: "Update from McHacks",
+                html: fs.readFileSync(path.join(__dirname, "../assets/email/waitlisted.html")).toString()
+            };
+            break;
+        default:
+            Services.Logger.error(`Invalid status change: ${req.body.status}`);
+    }
+    if(mailData) {
+        Services.Email.send(mailData).then(
+            (response) => {
+                if(response[0].statusCode !== 202) {
+                    next(response[0]);
+                } else {
+                    next();
+                }
+            }, next);
+    } else {
+        next({
+            status: 422,
+            message: "Invalid status change",
+            data: {
+                status: req.body.status
+            }
+        });
+    }
+}
 
+/**
+ * Updates a hacker that is specified by req.params.id, and then sets req.hacker to equal the updated
+ * hacker value. 
+ * @param {{params:{id: string}, body: *}} req 
+ * @param {*} res 
+ * @param {*} next 
+ */
+async function updateHacker(req, res, next) {
+    const hacker = await Services.Hacker.updateOne(req.params.id, req.body);
+    if(hacker) {
+        const acct = await Services.Account.findById(hacker.accountId);
+        if(!acct) {
+            return next({
+                status: 500,
+                message: "Error while searching for account by id when updating hacker",
+                data: {
+                    hackerId: hacker.id,
+                    accountId: hacker.accountId
+                }
+            });
+        }
+        req.email = acct.email;
```

Comment:
It may be a good idea to note that we're putting the email in req.email in case of success into the comments.

Reply:
Oh sorry! In an older version of this, I set req.hacker to be the updated value of hacker, but then changed it to just req.email. I'll change the comments to reflect that